typescript/web/src/components/import-button/import-images-modal/modal-dropzone/import-dropped-files/import-images.ts

```diff
@@ -0,0 +1,109 @@
+/* eslint-disable no-underscore-dangle */
+import { ApolloClient, gql } from "@apollo/client";
+import { v4 as uuidv4 } from "uuid";
+import Bluebird from "bluebird";
+import mime from "mime-types";
+import chunk from "lodash/fp/chunk";
+
+import { uploadFile } from "../../../../../utils/upload-file";
+import { DroppedFile, SetUploadStatuses } from "../../types";
+
+import { BATCH_SIZE, CONCURRENCY } from "../../constants";
+
+const createManyImagesMutation = gql`
+  mutation ($images: [ImageCreateManySingleImage!]!, $datasetId: ID!) {
+    createManyImages(data: { images: $images, datasetId: $datasetId }) {
+      id
+    }
+  }
+`;
+
+const uploadBatchOfImages = async ({
+  batch,
+  firstUploadDate,
+  batchIndex,
+  datasetId,
+  workspaceId,
+  apolloClient,
+}: {
+  batch: DroppedFile[];
+  firstUploadDate: Date;
+  batchIndex: number;
+  datasetId: string;
+  workspaceId: string;
+  apolloClient: ApolloClient<any>;
+}) => {
+  return await Promise.all(
+    batch.map(async ({ file }, fileIndex) => {
+      const createdAtDate = new Date();
+      createdAtDate.setTime(
+        firstUploadDate.getTime() + batchIndex * BATCH_SIZE + fileIndex
+      );
+      const createdAt = createdAtDate.toISOString();
+      const key = `${workspaceId}/${datasetId}/${uuidv4()}.${mime.extension(
+        file.type
+      )}`;
```

Comment:
What's the purpose of this key? Why is it different from the one in `import-datasets.tsx`

Reply:
To put and retrieve the image in the storage.

It's different that the one in import dataset because datasets imports need to upload a zip of json file to the storage, then this file is processed on the server side. 

As it's 2 different behaviours, I found it useful to add a difference between those 2 processes.